protocol.py

```diff
@@ -225,10 +226,20 @@ def start(self, master):
         def close(self):
             thread = self.thread
             if thread:
+                logger.debug('Thread')
                 self.thread = None
+
                 if self.httpd:
+                    logger.info('Shutting down httpd')
                     self.httpd.shutdown()
-                thread.join()	# Wait for it to quit
+
+                logger.info('Joining thread')
+                thread.join()  # Wait for it to quit
+
```

Comment:
```suggestion
logger.info('thread exited')

```

Reply:
This is covered by the end of function 'Done.'.